libs/core/langchain_core/language_models/chat_models.py

```diff
@@ -60,6 +63,15 @@
     from langchain_core.tools import BaseTool
 
 
+class LangSmithParams(TypedDict, total=False):
```

Comment:
this is kind of an abuse of TypedDict because I expect fields to be missing.

here I include all parameters that we are standardizing (including those that can be None). this is mostly just to define them somewhere.

regarding total=False: the implementation of _get_ls_params on BaseChatModel is almost the same as _get_invocation_params() (it just adds `ls_model_type="chat"`). so there should be minimal changes in tracing to models on which we haven't implemented _get_ls_params yet. but providers generally don't have the `ls_` params yet.